lib/spack/external/ctest_log_parser.py

```diff
@@ -136,6 +98,9 @@ def match(self, text):
     "^cc[^C]*CC: ERROR File = ([^,]+), Line = ([0-9]+)",
     "^ld([^:])*:([ \\t])*ERROR([^:])*:",
     "^ild:([ \\t])*\\(undefined symbol\\)",
+    "[^ :]{1,10} : (error|fatal error|catastrophic error)",
+    "[^:]{1,10}: (Error:|error|undefined reference|multiply defined)",
+    "[^:]{1,10}\\([^\\)]{1,128}\\) ?: (error|fatal error|catastrophic error)",
```

Comment:
```suggestion
    "[^:]{1,10}\\([^\\)]+\\) ?: (error|fatal error|catastrophic error)",
```
maybe this works too, since `(` and `)` are matched on both sides, so it can't suffer from the many-matches-problem